compiler/rustc_hir_analysis/src/check/region.rs

```diff
@@ -803,7 +861,7 @@ impl<'tcx> Visitor<'tcx> for ScopeResolutionVisitor<'tcx> {
                     local_id: body.value.hir_id.local_id,
                     data: ScopeData::Destruction,
                 });
-                resolve_local(this, None, Some(body.value));
+                resolve_local(this, None, Some(body.value), false);
```

Comment:
```suggestion
                resolve_local(this, None, Some(body.value), IsSuperLet::No);
```
hi, i am lcnr, a known boolean hater

Reply:
Changed the boolean to `LetKind::{Regular, Super}`.